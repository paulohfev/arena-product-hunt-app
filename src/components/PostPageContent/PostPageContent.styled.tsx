import Image from 'next/image'

import styled from 'styled-components'

export const PostPageWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 16px;
`

export const PostPageContentWrapper = styled.div<{ $direction: 'row' | 'column' }>`
  background-color: #ffffff;
  border-radius: 8px;
  display: flex;
  flex-direction: ${({ $direction }) => $direction};
  gap: 16px;
  padding: 24px;
`

export const PostPageBody = styled.div`
  display: flex;
  gap: 16px;
`

export const PostPageContentHeaderImage = styled(Image)`
  border-radius: 8px;
`

export const PostPageContentText = styled.div`
  display: flex;
  flex-direction: column;
  gap: 4px;
`

export const PostPageContentTitle = styled.h1`
  font-size: 24px;
`

export const PostPageContentDescription = styled.p`
  color: #666;
  font-size: 16px;
`

export const PostPageButtonsWrapper = styled.div`
  display: flex;
  justify-content: center;
  gap: 16px;
  width: 100%;
`

export const PostPageContentButton = styled.button`
  align-items: center;
  display: flex;
  background-color: #ffffff;
  border: none;
  border-radius: 16px;
  color: #fff;
  cursor: pointer;
  justify-content: center;
  padding: 10px;
  transition: all 0.2s;
  width: fit-content;

  &:hover {
    transform: translateY(-2px);
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
  }
`
