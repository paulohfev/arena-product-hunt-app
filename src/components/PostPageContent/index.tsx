import { useRouter } from 'next/navigation'

import { Post } from '@/types/Post'

import Button from '../Button'
import {
  PostPageBody,
  PostPageButtonsWrapper,
  PostPageContentDescription,
  PostPageContentHeaderImage,
  PostPageContentText,
  PostPageContentTitle,
  PostPageContentWrapper,
  PostPageWrapper,
} from './PostPageContent.styled'

export type PostPageContentProps = {
  post: Post
}

const PostPageContent: React.FC<PostPageContentProps> = ({ post }) => {
  const router = useRouter()

  return (
    <PostPageWrapper>
      <Button backgroundColor='white' border='1px solid #e0e0e0' color='black' onClick={() => router.back()}>
        Back
      </Button>

      <PostPageContentWrapper $direction='column'>
        <PostPageBody>
          <PostPageContentHeaderImage src={post.thumbnail.url} alt={post.name} width={64} height={64} />

          <PostPageContentText>
            <PostPageContentTitle>{post.name}</PostPageContentTitle>

            <PostPageContentDescription>{post.description}</PostPageContentDescription>
          </PostPageContentText>
        </PostPageBody>
      </PostPageContentWrapper>

      <PostPageContentWrapper $direction='row'>
        <PostPageButtonsWrapper>
          <Button backgroundColor='white' border='1px solid #e0e0e0' color='black' disabled>
            Get it
          </Button>

          <Button backgroundColor='#FF5733' color='white' disabled>
            Upvote ({post.votesCount})
          </Button>
        </PostPageButtonsWrapper>
      </PostPageContentWrapper>
    </PostPageWrapper>
  )
}

export default PostPageContent
