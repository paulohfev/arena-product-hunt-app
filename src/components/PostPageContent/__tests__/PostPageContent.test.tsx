import { render } from '@testing-library/react'

import PostPageContent from '../index'

jest.mock('next/navigation', () => ({
  useRouter: jest.fn().mockReturnValue({
    push: jest.fn(),
  }),
}))

const mockPost = {
  id: '1',
  name: 'Test Post',
  tagline: 'Test Tagline',
  description: 'Test Description',
  votesCount: 100,
  url: 'https://example.com',
  thumbnail: {
    url: 'https://example.com/image.jpg',
  },
}

describe('PostPageContent', () => {
  it('matches snapshot', () => {
    const { container } = render(<PostPageContent post={mockPost} />)
    expect(container).toMatchSnapshot()
  })
})
