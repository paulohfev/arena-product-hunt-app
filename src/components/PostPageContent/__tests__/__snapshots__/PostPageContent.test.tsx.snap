// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostPageContent matches snapshot 1`] = `
<div>
  <div
    class="sc-8fa4f82f-0 bxvMSM"
  >
    <button
      class="sc-8fa4f82f-8 bnPtyx"
    >
      <img
        alt="Back"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/_next/image?url=%2Ficons%2Fleft-arrow.png&w=48&q=75"
        srcset="/_next/image?url=%2Ficons%2Fleft-arrow.png&w=32&q=75 1x, /_next/image?url=%2Ficons%2Fleft-arrow.png&w=48&q=75 2x"
        style="color: transparent;"
        width="24"
      />
    </button>
    <div
      class="sc-8fa4f82f-1 jxdIQK"
    >
      <div
        class="sc-8fa4f82f-2 cJTwmm"
      >
        <img
          alt="Test Post"
          class="sc-8fa4f82f-3 fKezKH"
          data-nimg="1"
          decoding="async"
          height="64"
          loading="lazy"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=128&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=64&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fimage.jpg&w=128&q=75 2x"
          style="color: transparent;"
          width="64"
        />
        <div
          class="sc-8fa4f82f-4 kTtCcz"
        >
          <h1
            class="sc-8fa4f82f-5 fEJEZ"
          >
            Test Post
          </h1>
          <p
            class="sc-8fa4f82f-6 dIarqa"
          >
            Test Description
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-8fa4f82f-1 gQZcwy"
    >
      <div
        class="sc-8fa4f82f-7 fErWDg"
      >
        <button
          class="sc-4e78a862-0 eKfrzH"
          disabled=""
        >
          Get it
        </button>
        <button
          class="sc-4e78a862-0 kgCbZp"
          disabled=""
        >
          Upvote (
          100
          )
        </button>
      </div>
    </div>
  </div>
</div>
`;
